src/models/partial_tx.ts

```diff
@@ -0,0 +1,440 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import Input from './input';
+import Output from './output';
+import Address from './address';
+import Transaction from './transaction';
+import Network from './network';
+
+import P2PKH from './p2pkh';
+import P2SH from './p2sh';
+import ScriptData from './script_data';
+
+import transaction from '../transaction';
+import helpers from '../utils/helpers';
+import { IndexOOBError, UnsupportedScriptError } from '../errors';
+
+import txApi from '../api/txApi';
+import { TOKEN_AUTHORITY_MASK, TOKEN_INDEX_MASK, HATHOR_TOKEN_CONFIG } from '../constants';
+
+type InputData = {
+  tx_id: string,
+  index: number,
+  address?: string,
+  data: Buffer,
+};
+
+type OutputData = {
+  type: string,
+  value: number,
+  tokenData: number,
+  address?: string,
+  data?: string,
```

Comment:
This is for data outputs.

Reply:
But we don't have data outputs in atomic swap. Or the idea is to allow the user to create any type of transaction here?

Reply:
Maybe we could allow creating data outputs in the future.
But the actual reason is because I was following the "type" expected from the `createTxFromData` helper.